src/js/actions/layers.js

```diff
@@ -474,6 +475,7 @@ define(function (require, exports) {
                     };
                 });
                 if (amendHistory) {
+                    payload.suppressDirty = true;
```

Comment:
This makes sense.  In fact, in theory, since the other code path here doesn't even *touch* history, I would expect that you could get away with ALL resetLayers suppressing dirty.  That said, I do think this change makes me a bit nervous.